server/taskflows/hpccloud/taskflow/nwchem/__init__.py

```diff
@@ -210,12 +214,56 @@ def monitor_nwchem_job(task, upstream_result):
     task.throws=(Retry,),
 
     job = upstream_result['job']
-    # TODO - We are currently reaching in and used a 'private' function
     _monitor_jobs(task, cluster, [job], girder_token=girder_token, monitor_interval=30)
 
     return upstream_result
 
 @cumulus.taskflow.task
+def create_json_output(task, upstream_result):
+    task.logger.info('Converting nwchem output to json format.')
+    cluster = upstream_result['cluster']
+    job = upstream_result['job']
+    job_dir = job_directory(cluster, job)
+    out_file = '%s.out' % (job['name'])
+
+    try:
+        # Copy the nwchem output to server
+        tmp_dir = tempfile.mkdtemp()
+        cluster_path = os.path.join(job_dir, out_file)
+        local_path = os.path.join(tmp_dir, out_file)
+        with open(local_path, 'w') as local_fp:
+            with get_connection(task.taskflow.girder_token, cluster) as conn:
+                with conn.get(cluster_path) as remote_fp:
+                    local_fp.write(remote_fp.read())
+
+        # Run docker container to post-process results - need to add docker image to upstream_result
+        command = ['docker', 'run', '--rm', '-v', '%s:/hpccloud' % tmp_dir,
+                'chetnieter/nwchem-postprocess', out_file]
+        p = subprocess.Popen(args=command, stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE)
+        stdout, stderr = p.communicate()
+
+        if p.returncode != 0:
+            print('Error running Docker container.')
+            print('STDOUT: ' + stdout)
+            print('STDERR: ' + stderr)
+            raise Exception('Docker returned code {}.'.format(p.returncode))
+
+        # Copy json file back to cluster?
+        cluster_path = os.path.join(job_dir, out_file + '.json')
+        local_path = os.path.join(tmp_dir, out_file + '.json')
+        with get_connection(task.taskflow.girder_token, cluster) as conn:
+            with open(local_path, 'r') as local_fp:
+                conn.put(local_fp, cluster_path)
```

Comment:
We don't need the JSON back on the cluster, it should just be uploaded to Girder